lib/models/os.coffee

```diff
@@ -35,29 +41,27 @@ OSParams = require('./os-params')
 # parameters =
 #		network: 'ethernet'
 #		appId: 91
+#		onProgress: (state) ->
+#			return if not state?
+#			console.log "Total: #{state.total}"
+#			console.log "Received: #{state.received}"
 #
 # resin.models.os.download parameters, '/opt/os.zip', (error) ->
 #		throw error if error?
-#	, (state) ->
-#		return if not state?
-#		console.log "Total: #{state.total}"
-#		console.log "Received: #{state.received}"
 ###
-exports.download = (parameters, destination, callback, onProgress) ->
+exports.download = (options, destination, callback) ->
+	Promise.try ->
+		if not token.getUsername()?
+			throw new errors.ResinNotLoggedIn()
 
-	if not token.getUsername()?
-		return callback(new errors.ResinNotLoggedIn())
+		osParams = _.omit(options, 'onProgress')
```

Comment:
It would make more sense to whitelist the options you expect rather than blacklist specific ones you don't expect (because they could always pass an option that is invalid under any context, rather than a valid one that just isn't expected for this branch)